frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -630,24 +631,31 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 				return nil, parser.WithLocation(err, d.stage.Location)
 			}
 		}
+
 		d.state = d.state.Network(opt.NetworkMode)
+		dockerIgnorePatterns, err := opt.Client.DockerIgnorePatterns(ctx)
```

Comment:
Although the return of this function seems to be cached, it is still weird that this happens inside `for` loop for each stage. I think it should be moved out as it is not stage-dependent.